actors/miner/src/lib.rs

```diff
@@ -1950,6 +1951,238 @@ impl Actor {
+                return Err(actor_error!(
+                    illegal_argument,
+                    "seal challenge epoch {} must be before now {}",
+                    sector.seal_rand_epoch,
+                    curr_epoch
+                ));
+            }
+
+            if sector.seal_rand_epoch < challenge_earliest {
+                return Err(actor_error!(
+                    illegal_argument,
+                    "seal challenge epoch {} too old, must be after {}",
+                    sector.seal_rand_epoch,
+                    challenge_earliest
+                ));
+            }
+        }
+
+        let state: State = rt.state()?;
+        let store = rt.store();
+        let miner_id = rt.message().receiver().id().unwrap();
+        let info = get_miner_info(rt.store(), &state)?;
+        let activation_epoch = rt.curr_epoch();
+
+        if consensus_fault_active(&info, rt.curr_epoch()) {
+            return Err(actor_error!(
+                forbidden,
+                "prove-commit-ni not allowed during active consensus fault"
+            ));
+        }
+
+        rt.validate_immediate_caller_is(
+            info.control_addresses.iter().chain(&[info.worker, info.owner]),
+        )?;
+
+        let base_sector_power = base_power_for_sector(info.sector_size);
+
+        let rew = request_current_epoch_block_reward(rt)?;
+        let pwr = request_current_total_power(rt)?;
+        let circulating_supply = rt.total_fil_circ_supply();
+        let pledge_inputs = NetworkPledgeInputs {
+            network_qap: pwr.quality_adj_power_smoothed,
+            network_baseline: rew.this_epoch_baseline_power,
+            circulating_supply,
+            epoch_reward: rew.this_epoch_reward_smoothed,
+        };
+
+        let initial_pledge = initial_pledge_for_power(
+            &base_sector_power,
+            &pledge_inputs.network_baseline,
+            &pledge_inputs.epoch_reward,
+            &pledge_inputs.network_qap,
+            &pledge_inputs.circulating_supply,
+        );
+
+        let day_reward = expected_reward_for_power(
+            &pledge_inputs.epoch_reward,
+            &pledge_inputs.network_qap,
+            &base_sector_power,
+            fil_actors_runtime::EPOCHS_IN_DAY,
+        );
+
+        let storage_pledge = expected_reward_for_power(
+            &pledge_inputs.epoch_reward,
+            &pledge_inputs.network_qap,
+            &base_sector_power,
+            INITIAL_PLEDGE_PROJECTION_PERIOD,
+        );
+
+        let sector_verify_info = params
+            .sectors
+            .iter()
+            .zip(params.sector_proofs.into_iter())
+            .map(|(sector, proof)| SealVerifyInfo {
+                registered_proof: params.seal_proof_type,
+                sector_id: fvm_shared::sector::SectorID {
+                    miner: miner_id,
+                    number: sector.sector_number,
+                },
+                randomness: Randomness(Vec::new()),
```

Comment:
This one needs to be initialised properly, though the interactive_randomness below doesn't.

Reply:
Actually, even `interactive_randomness` should be initialized to commr right? https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0090.md#specification

Reply:
I would have expected that to happen inside the proofs library. I think we need to ask someone else.

Reply:
It's internally ignored, so it can be the same as comm_r.  I believe if you pass all zeroes, it should fail as that's marked as invalid.